# README

## Purpose and overview:

The purpose of this program is to take in a file of k-mers and develop an optimal sequence derived from a De Bruijn graph. Before creating the De Bruijn graph, the k-mers are first decontaminated by known vectors and then any infrequent sub k-mers are then replaced with more frequent k-mers to ensure the integrity of the DNA samples that are given. Once all the k-mers are decontaminated and corrected, the k-mers are placed in a De Bruijn algorithm that finds the longest sequence for all the k-mers. The program was able to take 50-mer reads and reconstruct 16 reads of length 1260 of the Sars-Covid-2 receptor binding domain. 

## How to run the program:

There are 2 shell script files titled debruijn.sh and assembly.sh.

### debruijn.sh 
To run the debruijn script, run the script in the terminal `$ sh debruijn.sh <reads.txt>` where reads.txt is a text file containing a list of DNA reads of constant length (separated by newlines). Example: `sh debruijn.sh test_cases/debruijn/reads2.txt`. The shell script calls a python file that reads in k-mers, convert them into the edges of an internal representation of a de Bruijn graph, simplify this internal representation, and finally print out all possible most likely (i.e., maximum-length) target sequences which can be inferred from the input reads. The program also generates a DOT file called debruijn.dot representing the fully simplified de Bruijn graph. The program simplifies the debruijn graph where every “singleton” is merged into one node and the edges in the resulting graph are updated to reflect the full overlap of the final node labels. 

Example debruijn graph:

<img src="https://github.com/AlexKaiLe/Genome-Assembly/blob/main/images/debruijn.png" alt="eq" height="600"/>


### assembly.sh
To run the assembly script, run the script in the terminal `$ sh assembly.sh <reads.txt> <vector.txt> <contam-k> <k-mer length> <k> <t> <d>`. Example: `sh assembly.sh test_cases/assembly/sampleReads.txt test_cases/assembly/vector.txt 10 30 17 2 2`. The shell script calls a python file that reads in k-mers from reads and decontaminates the reads with a given vector. The program finds the longest contig of these sequences and returns that sequence.

Output of assembly can be found at the bottom of the file.


## Contamination:
Takes in 3 parameters
- The contaminated reads
- The vector
- The k value to create kmers

This file runs very similar to BLAST where there is a seeding and extension phase. However, this only happens at the begining and end of the contaminated read. Thus, we only seed at the end, then extend in the corresponding direction of both the contaminated read and the vector. Once it reaches a comparison that does not match, we terminate. We return the new read that removes all the contamination.


## Correction:
Takes in 4 parameters
- The error reads
- k = kmer length
- t = threshold for frequent kmer
- d = hamming distance threshold

First we find a list of the most frequent k-mers. Then we run through all the reads and generate kmers from each of the read to find the uncommon kmers. Once we find all the uncommon kmers, we try to find the best kmer from the frequent kmer list to replace them using the distance variable d. Once we run through all unfrequent kmers in the current read and find the best corresponding repalcement (sorted by distance values and then by frequency), we look at how many uncommon kmers a replacement will create if an infrequent k-mer is replaced. From all the possible k-mers we can replace, we want choose to replace the uncommon kmer that creates the least amount of infrequent k-mers after it is replaced. In doing this, we optimize the replacement of all the infrequnet k-mers in a sequence. We the loop again to create more kmers of the new seqeunce and see if we can replace any more infrequent kmers. I set a max on the number of loops to be 5 because it has a very large runtime (ie. the application runtime is veryyyyy long). Once we replace all the infrequent kmers as well as the infrequent kmers that may have be generated from previous placements, we print out the new reads for a specific sequence. I only use the frequent kmer list when I first created it because it adding new kmers to that list could "infect" the rest of the seqeunces with a frequent kmer that never existed in the orgianl sequence. I also chose to use the sequence that causes the least amount of new infrequent kmers to replace the orginal sequence so that we can optimize the number of frequent kmers that do exist in the seqeunce. 



## Workflow:
I took 3 files and combined them in sequential order to find the final inferred sequence of a sequence of various k-mers. I first took the reads files and removed any of the reads which contained contamination. I did this by running all the reads through the contamination.py file, which returns a list of contaminated reads. Once I removed all the contaminated reads, I determined the reads which I needed to correct. Because I do not know the actual length of the fully sequenced sampleRead.txt, I have to take in all the kmers derived from the reads to sequence to find the longest contig. I place decontaminated reads into the correction.py file, which replaces all infrequent kmers with frequent kmers throughout the reads. Now I have a list of decontaminated and corrected reads. Afterward, I generate kmers from this list of a specified length. These kmers are then placed into the debruijn.py file that then generates the longest contiguous segment that can be inferred from these kmers.

I chose to do contamination before correction because I do not want to be correcting a contamination region. If I corrected a contamination region by accident, then our contamination algorithm would not be able to determine if the read was contaminated or not because that region would have changed as a result of the correction. It is possible that a contamination region may have been altered and thus, preventing the algorithm from detecting the contamination. In this case, using correction to rectify this alteration would be helpful. However, this case is more unlikely to occur and it would prove to be more harmful if I accidentally introduced more alterations in the contamination regions that would prevent the algorithm from identifying it and removing it. 

At the end of the correction phase, I pass all the corrected reads into the de Bruijn file to create the graph and find the longest contig that can be inferred. 

For the sampleReads.txt file, I used the parameters [10, 30, 17, 2, 2]. 
- 10 = kmer length to determine seeding in contamination 
- 30 = kmer length of reads being used to correct and construct a De Bruijn graph 
- 17 = kmer length to determine seeding in correction 
- 2 = threshold value to determine the frequency of common kmer values in correction
- 2 = difference value for how much a replaceable kmer can differ by in correction

## Detecting Errors by optimizing variables for k (k-mer length) and t (threshold for frequent k-mers)

Analyze the hiv-1_genome.txt that contains 9,181 bp RNA genome of HIV-1using unitary.m, which is a DNA scoring matrix. Simulate extraction of 50-mer reads from the HIV genome  where I mutated these reads randomly based on a 1% error rate

P = probability of number of reads covering a given position of the sequence
P=(e^(-a)a^k)/k! where a = N/(LG)
To find the min number of reads covering the entire sequence 
- K = 0 → no reads cover the entire sequence 
- L = 50 → length of every read
- G = 9181 → length of the entire sequence
- N = ? → The number of reads covering the entire length of the sequence 
- 1-P = 1 - the probability that no reads cover the entire sequence = 0.99

Ensure that at least 99.9% of the target sequence is expected to be covered
1 - (e^(-a)a^k)/k!= 0.999
 
- 1 - (e^(-a)a^0)/0! = 0.999
- 1 -e^(-a) = 0.999
- 1 e^(-(NL)/G) = 0.999
- 0.001 =e^(-(NL)/G)
- ln(0.001) = -(NL)/G)
- (-G • ln(0.001)) / L = N
- (-9181 • ln(0.001)) / 50 = 1268.4 ≈ 1269

Thus, 1269 reads of 50-mers to ensure that 99.9% of the target sequence is expected to be covered

Values of 
- Nmin = 1269
- Sm = 49.981087470449175
- max(Sk) = 49.99211977935382 where k = 22, t = 4
- max(St) = 50 where k = [17-25], t = 2

- average ungapped global alignment score of each true/mutated = S<sub>M</sub>
- average ungapped global alignment score of each true/corrected read pair = S<sub>K</sub> =
<img src="https://github.com/AlexKaiLe/Genome-Assembly/blob/main/images/eq_1.png" alt="eq" height="200"/>

<img src="https://github.com/AlexKaiLe/Genome-Assembly/blob/main/images/1.png" alt="eq" height="300"/>

<img src="https://github.com/AlexKaiLe/Genome-Assembly/blob/main/images/2.png" alt="eq" height="300"/>


note: can’t make St values 50 because it would be log(0) = undefined, thus any 50 values was turned into 49.999 to graph it

## General trends observed in varying k and t
For all combinations of K and T, we see a downward trend for the similarity score for lower amounts of K (ie. the kmers are smaller). However, when we increase the value of K, we see an increase in accuracy of similarity score for all values of T. In the graph, we also see that with lower K values, lower T values will result in an increase in the corrected results. However, at higher K values, a higher T value results in an increase in the similarity score. The only outlier to these cases is when T = 4 where it has the highest similarity score at higher values of K despite having a small T.

These trends can also be observed in the second graph. However, because values below T=7 are optimized based on the coverage, we see that for all lower T values, the similarity score will increase despite the value of K. This makes sense for optimized values of T because having a lower threshold value will allow for more frequent kmers to replace the infrequent k-mers. A further explanation can be found in question 3.
These results make sense because at lower thresholds of T, there will be more frequent k-mer to replace with. However, at higher thresholds of T, there will be less frequent k-mer to replace with and where these frequent k-mers are more accurate since they appear more often in the sequence.  
At lower values of k, we see that lower values of T have a higher similarity score to the true read. Since many kmers are generated at a lower k value, infrequent k-mers can be easily replaced. Thus, it is easier to surpass the T threshold and will create a bigger list of frequent kmers. As a result, lower T values at low K values will ensure that a more correct k-mer will be used to replace the infrequent kmers. This is because the list of the frequent kmers is large and we will be able to find a kmer that does not differ too much. Higher T values at low K values will have more incorrect k-mer being used to replace the infrequent kmers since the list of the frequent kmers is smaller and may not include the optimal replacement kmer.  

At higher values of k, we see that lower values of T have a low similarity score to the true read. Since only a few kmers are generated at a high k value, infrequent k-mers will not be easily replaced. Thus, it is harder to surpass the T threshold and will create a smaller list of frequent kmers. As a result, higher T values at high K values will ensure that a more correct k-mer will be used to replace the infrequent kmers. This is because we will have a small list of frequent k-mer that are really long and having a bigger threshold will allow more matches to be made. On the other hand lower T values at high K values will not have many k-mers available to replace infrequent kmers since the list of the frequent kmers will be smaller due to a low T and the nature of the long sequence with a higher k. Thus, the most frequent kmer list may not include the optimal replacement kmer or any kmers to replace at all. 

It seems like if we continue to increase T, we would get a more optimal similarity score to the true read at higher k values. However, we can see with T = 12 that even with a high T value, there are no k values that reach a similarity score greater than 0. This is because T = 12 will have the smallest most frequent list and as we keep increasing the K-mer length, it will become increasingly harder to find k-mers that best match the infrequent kmer. Thus, we will have a plateau since we can not perform any better than the mutated read. 

However, when T = 4, we see that it outperforms all the other T values despite altering the kmer length. This must be because T  values around 4 will account for the mutation rate of HIV (0.001) and thus is the optimal threshold for which HIV is mutated at. Because the coverage is a=(NL)/G= 50•1269)/9181 = 6.911 ≈ 7 , a mutation rate may cause the coverage of some nucleotides to be in the range of 4-7 coverage. Although T = 6 lies within this range, T = 4 is most optimal to account for the mutation where it allows for more kmers to be added to the frequent kmer list. Thus, it is shown that T = 4 is most optimal of the current list of T values. (explained more in question 3) 

## Choice of t dependent on k

At low values of k, we create a large list of kmers and thus, each kmer has a relatively high frequency where there is little difference between each k-mer. At high values of k, we create a smaller list of kmers and thus, each kmer has a relatively low frequency where there is a bigger difference between each k-mer. Larger kmers result in higher similarity score because they span a greater distance and have more indices to match when calculating the replaceable “frequent” k-mers. Furthermore, the number of frequent k-mers for a bigger k holds more importance since the k-mer is a longer sequence and needs to be matched exactly with another k-mer which statistically is hard to do with larger k values. Thus, when we replace an infrequent k-mer, we know that the frequent k-mer will most likely be more accurate for higher K values. As a result, we want to choose T values that optimize the similarity score when K is larger. In our case with HIV, we want to choose the T value that has the highest similarity score in the range 15-25. 

![](https://github.com/AlexKaiLe/Genome-Assembly/blob/main/images/3.png)

There is only one area where varying T for the same K does not produce a significant difference in correction efficiency. However, the majority of the similarity scores in general do not converge on a single point. The smallest difference in the 5 similarity scores for each T value happens around K = 25. The reason why the similarity scores converge on around this area for K values is because when K is really large, the number of frequent K-mers decreases. We have less frequent K-mers because K spans a large distance on the sequence and it is very hard to match these large sequences together. Thus, the number of frequent k-mers we can use to replace the infrequent k-mers is small. Since our d value is 2, the infrequent k-mer must only differ by 2 for frequent k-mer to replace it and with such a large k-mers, there are very few or no frequent k-mer that we can use to replace the infrequent k-mers with. Thus, with very few replacements as a result of the large k value, all the similarity scores converge near to 0, which means that it is similar to the mutated sequence. 

## Features of the data are the parameters k and t dependent on variables

Some of the features K and T are dependent on is the number of reads, the length of each read, and the length of the input sequence. These features help determine the coverage of a single nucleotide within a sequence. The coverage is calculated by  a=(NL)/G = 50 • 1269/9181 = 6.911 ≈ 7 . In our case with HIV, our coverage is 7, which means that for any given nucleotide, it will be covered by 7 separate sequences. The value a (which equals 7 in our case) is a function of the length of a sequence, the length of each read and the number of reads, the value of K and T that is most optimal will be determined by these parameters. We should expect to see T values below 7 to perform better because it is likely that the k-mer will appear that many times (i.e. value below than 7) in the original sequence that we can replace in our algorithm. The value of K will then be based on the threshold of T. This is because the average frequency of k-mers will decrease as K increases since k-mers become more unique as K increases. 

I would modify my code to take advantage of this dependence by using the coverage value of a (ie. 7) to find a set of optimal values for K and T using the number of reads, the length of each read, and the length of the input sequence. Instead of using random values of K and T to find the optimal values, we can use these parameters to refine our algorithm search and thus, reduce our runtime. For example, since we know the coverage value, all we need to do is make the upper bound of T be the coverage. Then we can choose a set of K values that is optimal for the genetic diversity of the sequence. If a sequence has little diversity, our K values should be relatively large in order to encapsulate the uniqueness. However if a sequence has a lot of diversity, the K values do not need to be very large because the k-mers will encapsulate the uniqueness of the sequence with a shorter read. This will make the expected accuracy of the algorithm higher because we are using optimized parameters. 

## Strengths

A strength is that I implement various algorithms to remove contamination and mutations to generate reads that are most likely the correct values of the original genome. By cleaning up the reads before inputting them into the de Bruijn graph, I reduce the probability of a kmer that will introduce errors into the final inferred sequence. Another strength is that both contamination and mutations use BLAST-like seeding and extension to find the optimal reads. Using this technique allows for the best reads to be produced. The de Bruijn graph also adds another layer of correction because I only produce the longest reads that can be inferred from the graph. If there happens to be a kmer that is incorrect that has become a node within the graph, it is very likely that it will be part of the inferred sequence because it will be an uncommon node with very few incoming/outgoing edges. Finally, I have tuned the parameters of the assembly.py file to ensure that optimal reads and contigs are produced. More about these parameters can be found in the intermediate statistics section.

## Shortcomings

A shortcoming is run time. The two longest sections that contribute to a long runtime are correction and de Bruijn. Because I generate so many kmers that need to be corrected, the runtime in which correction performs is very large. Similarly, because I generate a ton of kmers that act as nodes, it takes a long time for the program to perform a depth-first search in finding all the contigs. Another shortcoming is that I do not know if this program is actually getting a sequence close to that of the original genome because I have nothing to compare to. I am using numbers that I believe can obtain this information, but there is no way to truly know if the program is optimally working under the parameters I am using. Another limitation to this program is that I am only using one de Bruijn graph to infer the longest contig. A better implementation would be to create several de Bruijn graphs with different parameters (ie. different lengths of kmers) so that I can gather more information about the sequence. Finally, the program produces a variety of contigs of equal length. As a result, I do not know which contig best represents the true sequence. 

## Intermediate statistics:
- Total number of reads: 1274
- Number of contaminated reads: 51
- Number of uncontaminated: 1223
- Number of frequent k-mers: 1613 / 8498 = 0.18980936690986114
- Number of corrected reads: 1223

A lot of the parameter tuning came from trial and error
- 10 = kmer length to determine seeding in contamination 
    - The length of the contamination kmer value had to be big enough to ensure that the vector does indeed contaminate the reads. However, it can not be too big, or else it would be very unlikely that I will find an exact match for the vector in our reads. If the kmer is too small, all the reads will be marked as contaminated because it is more likely that I can find a smaller read that exactly matches the vector than a larger kmers. Thus, to balance these two, I arrived at the number 10. 
- 30 = kmer length of reads being used to correct and construct a De Bruijn graph 
    - I want a kmer length that results in a lot of connections in a de Bruijn graph, but also represents a long sequence of correct reads. If I had a kmer value that was too small, it would result in a very large graph and would be super complex to traverse with lots of nodes and edges. On the other hand, if I had a large kmer value, it would result in a very small and simple graph. Thus to balance these two bounds, I decided to have a value in the middle and as a result, I arrived at the number 30. 
- 17 = kmer length to determine seeding in correction 
    - The length of the kmers derived from the reads should be similar in length to that of the optimal kmer length I found in pr2. In pr2, I saw that as I increased the kmer length to around 17-20, the higher the accuracy of the true genome was. A larger kmer means that is a lower frequency of that kmer. As a result, any kmer that surpasses a frequency threshold will most likely be a correct kmer of the true read. Thus, to get as close to the genome I am trying to sequence, I should use a number that is similar, which is why I arrived at the number 17. I should not use too large of a number because that may cause no frequent kmers and thus, I have no kmers to replace. 
- 2 = threshold value to determine the frequency of common kmer values in correction
    - The threshold value should also be similar in length to that of the optimal kmer length I found in pr2. In pr2, I saw that as I decreased the threshold value to around 2-4, the higher the accuracy of the true genome was. A smaller threshold means that there are more frequent kmers and less infrequent kmers. As a result, there are a lot more options to choose from to replace the infrequent kmers and will most likely result in the correct sequence. Thus, to get as close to the genome I are trying to sequence, I should use a number that is similar, which is why I arrived at the number 2. 
- 2 = difference value for how much a replaceable kmer can differ by in correction
    - The difference value should be a small number because I want to be as close to the true sequence as possible. If the difference value was super large, I could replace the kmer with a lot more possible kmers that may be incorrect. However, if I had a really small difference value, I would only have a few options to choose from. These options are most likely to be optimal kmers that can replace the incorrect kmer because they are derived from the most frequent kmer list and are very similar to the current kmer. This indicates that the new kmer used to replace is commonly found throughout the genome and is likely to be correct. Thus, I arrived at the number 2. 

## Findings: 
- After applying the De Bruijn algorithm to the kmer reads, I am able to obtain 16 reads of length 1260. An example reads i provided below 

- ATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGCTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGAGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTGCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCATACAATGTAACACAAGCTTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCAGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTACACAGGTGCCATCAAATTGGATGACAAAGATCCAAATTTCAAAGATCAAGTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACTCAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAA
- ATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGCTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGAGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTGCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCATACAATGTAACACAAGCTTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCCGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTACACAGGTGCCATCAAATTGGATGACAAAGATCCAAATTTCAAAGATCAAGTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACTCAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAA
- ATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGCTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGAGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTTCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCATACAATGTAACACAAGCTTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCAGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTACACAGGTGCCATCAAATTGGATGACAAAGATCCAAATTTCAAAGATCAAGTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACTCAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAA
- ATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGCTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGAGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTTCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCATACAATGTAACACAAGCTTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCCGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTACACAGGTGCCATCAAATTGGATGACAAAGATCCAAATTTCAAAGATCAAGTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACTCAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAA
- ATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGCTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGCGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTGCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCATACAATGTAACACAAGCTTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCAGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTACACAGGTGCCATCAAATTGGATGACAAAGATCCAAATTTCAAAGATCAAGTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACTCAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAA
- ATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGCTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGCGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTGCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCATACAATGTAACACAAGCTTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCCGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTACACAGGTGCCATCAAATTGGATGACAAAGATCCAAATTTCAAAGATCAAGTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACTCAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAA
- ATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGCTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGCGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTTCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCATACAATGTAACACAAGCTTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCAGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTACACAGGTGCCATCAAATTGGATGACAAAGATCCAAATTTCAAAGATCAAGTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACTCAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAA
- ATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGCTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGCGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTTCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCATACAATGTAACACAAGCTTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCCGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTACACAGGTGCCATCAAATTGGATGACAAAGATCCAAATTTCAAAGATCAAGTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACTCAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAA
- ATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGTTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGAGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTGCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCATACAATGTAACACAAGCTTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCAGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTACACAGGTGCCATCAAATTGGATGACAAAGATCCAAATTTCAAAGATCAAGTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACTCAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAA
- ATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGTTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGAGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTGCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCATACAATGTAACACAAGCTTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCCGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTACACAGGTGCCATCAAATTGGATGACAAAGATCCAAATTTCAAAGATCAAGTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACTCAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAA
- ATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGTTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGAGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTTCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCATACAATGTAACACAAGCTTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCAGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTACACAGGTGCCATCAAATTGGATGACAAAGATCCAAATTTCAAAGATCAAGTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACTCAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAA
- ATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGTTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGAGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTTCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCATACAATGTAACACAAGCTTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCCGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTACACAGGTGCCATCAAATTGGATGACAAAGATCCAAATTTCAAAGATCAAGTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACTCAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAA
- ATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGTTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGCGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTGCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCATACAATGTAACACAAGCTTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCAGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTACACAGGTGCCATCAAATTGGATGACAAAGATCCAAATTTCAAAGATCAAGTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACTCAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAA
- ATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGTTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGCGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTGCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCATACAATGTAACACAAGCTTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCCGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTACACAGGTGCCATCAAATTGGATGACAAAGATCCAAATTTCAAAGATCAAGTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACTCAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAA
- ATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGTTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGCGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTTCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCATACAATGTAACACAAGCTTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCAGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTACACAGGTGCCATCAAATTGGATGACAAAGATCCAAATTTCAAAGATCAAGTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACTCAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAA
- ATGTCTGATAATGGACCCCAAAATCAGCGAAATGCACCCCGCATTACGTTTGGTGGACCCTCAGATTCAACTGGCAGTAACCAGAATGGAGAACGCAGTGGGGCGCGATCAAAACAACGTCGGCCCCAAGGTTTACCCAATAATACTGCGTCTTGGTTCACCGCTCTCACTCAACATGGCAAGGAAGACCTTAAATTCCCTCGAGGACAAGGCGTTCCAATTAACACCAATAGCAGTCCAGATGACCAAATTGGCTACTACCGAAGCGCTACCAGACGAATTCGTGGTGGTGACGGTAAAATGAAAGATCTCAGTCCAAGATGGTATTTCTACTACCTAGGAACTGGGCCAGAAGCTGGACTTCCCTATGGTGCTAACAAAGACGGCATCATATGGGTTGCAACTGAGGGAGCCTTGAATACACCAAAAGATCACATTGGCACCCGCAATCCTGCTAACAATGCTGCAATCGTGCTACAACTTCCTCAAGGAACAACATTGCCAAAAGGCTTCTACGCAGAAGGGAGCAGAGGCGGCAGTCAAGCCTCTTCTCGTTCCTCATCACGTAGTCGCAACAGTTCAAGAAATTCAACTCCAGGCAGCAGTAGGGGAACTTCTCCTGCTAGAATGGCTGGCAATGGCGGTGATGCTGCTCTTGCTTTGCTGCTGCTTGACAGATTGAACCAGCTTGAGAGCAAAATGTCTGGTAAAGGCCAACAACAACAAGGCCAAACTGTCACTAAGAAATCTGCTGCTGAGGCTTCTAAGAAGCCTCGGCAAAAACGTACTGCCACTAAAGCATACAATGTAACACAAGCTTTCGGCAGACGTGGTCCAGAACAAACCCAAGGAAATTTTGGGGACCAGGAACTAATCAGACAAGGAACTGATTACAAACATTGGCCGCAAATTGCACAATTTGCCCCCAGCGCTTCCGCGTTCTTCGGAATGTCGCGCATTGGCATGGAAGTCACACCTTCGGGAACGTGGTTGACCTACACAGGTGCCATCAAATTGGATGACAAAGATCCAAATTTCAAAGATCAAGTCATTTTGCTGAATAAGCATATTGACGCATACAAAACATTCCCACCAACAGAGCCTAAAAAGGACAAAAAGAAGAAGGCTGATGAAACTCAAGCCTTACCGCAGAGACAGAAGAAACAGCAAACTGTGACTCTTCTTCCTGCTGCAGATTTGGATGATTTCTCCAAACAATTGCAACAATCCATGAGCAGTGCTGACTCAACTCAGGCCTAA
